src/api/web/endpoint/package.ts

```diff
@@ -70,6 +71,10 @@ function addPackageWebApi(route: Router, storage: IStorageHandler, auth: IAuth,
             logger.logger.error({ name: pkg.name, error: err }, 'permission process for @{name} has failed: @{error}');
             throw err;
           }
+
```

Comment:
Unfortunately, this won't work as you expect.  First, the `pkgCopy` is checking permissions above, so, whatever action is needed it must be within the `try/catch`. Furthermore, `dist` is a property that belongs to `Version` rather `Package`.

Reply:
@juanpicado I've tested it. Packages gotten from storage do have dist property, and i'm not sure if they do have a version.